File: week 12/leetcode/maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int count=0,  maxi = 0, n = s.size(); 
        int mini = 0, j =0; 
        char first = s[j]; 
        for(int i =0;i < n ;i++) {
            if(mini < k) {
            if(s[i] == 'a' or s[i] == 'e' or s[i] == 'i' or s[i] == 'o' or s[i] == 'u' ){
                mini++;
            
                count++;}
            else { 
                mini++; 
            }
            }
         else  if(mini == k) {
                maxi = max(count , maxi); 
                first = s[j];  
                if(first == 'a' or first== 'e' or first == 'i' or first  == 'o' or first == 'u') count--; 
                if(s[i]  == 'a' or s[i] == 'e' or s[i] == 'i' or s[i] == 'o' or s[i] == 'u') count++; 
                j++; 
            }
        }
        maxi = max(count, maxi); 
        return maxi; 

    }
};

File: week 12/leetcode/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        count = 0
        maxi = 0
        mini = 0
        j = 0
        first = s[j]
        n = len(s)
        for i in range(n):
            if mini < k:
                if s[i] in ['a', 'e', 'i', 'o', 'u']:
                    mini += 1
                    count += 1
                else:
                    mini += 1
            elif mini == k:
                maxi = max(count, maxi)
                first = s[j]
                if first in ['a', 'e', 'i', 'o', 'u']:
                    count -= 1
                if s[i] in ['a', 'e', 'i', 'o', 'u']:
                    count += 1
                j += 1
        maxi = max(count, maxi)
        return maxi
